### ISSUE TITLE
[9.0.0+] Adding HyperFormula leads to "TypeError: Converting circular structure to JSON"

### ISSUE THREAD

**Issue by GlennMatthys** [handsontable/handsontable#8728]
### Description
When the HyperFormula instance is added to the settings of a hot-table component, a warning from Vue is emitted about a circular JSON structure:

```
[Vue warn]: Error in getter for watcher "mergedHotSettings": "TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'DependencyGraph'
    |     property 'lazilyTransformingAstService' -> object with constructor 'LazilyTransformingAstService'
    |     property 'undoRedo' -> object with constructor 'UndoRedo'
    |     property 'operations' -> object with constructor 'Operations'
    --- property 'dependencyGraph' closes the circle"

found in

---> <HotTable>
```

### Steps to reproduce
See jsfiddle

### Demo
Clicking "Run" hangs the browser for a while until it errors out. https://jsfiddle.net/z3568ovc/ The problem is the same but the error message is different because Vue is not running with development mode on in jsfiddle.

```
<a class='gotoLine' href='#"[Vue warn'>"[Vue warn</a>: You may have an infinite update loop in watcher with expression \&quot;settings\&quot;

found in

---&gt; &lt;HotTable&gt; at src/common/HotTable.vue
       &lt;Root&gt;"]
```

### Your environment
* Vue wrapper version: 9.0.2
* Handsontable version: 9.0.2
* Browser Name and version: Google Chrome Version 93.0.4577.82 (Official Build) (64-bit)
* Operating System: Windows 10 Pro Version 2004 (OS Build 19041.1237)

**Comment by AMBudnik** [comment]
Thank you for the report @GlennMatthys 

Can you help me to test it out @aninde ?

**Comment by fsnlarson** [comment]
We have this issue as well

**Comment by sequba** [comment]
The provided demo uses the old vue wrapper (`https://cdn.rawgit.com/handsontable/vue-handsontable-official/40abf2c4/dist/ce/vue-handsontable-ce.min.js`).

I created a [jsfiddle ](https://jsfiddle.net/a4dpcsxn/14/) with current versions of `handsontable`,  and `@handsontable/vue`. It seems to work (displays the table, allows to type formulas and calculates them), but there's still an error in the console:
![image](https://user-images.githubusercontent.com/2480086/172415997-129d3006-7a03-4a59-8fcb-de12f275b721.png)

Anyway the issue should be fixed in vue wrapper

- If the cause of this error is the same as before (`Converting circular structure to JSON`), it is probably due to incorrect way of rendering table by the wrapper. [This stackoverflow question](https://stackoverflow.com/questions/56508997/vue-warn-error-in-render-typeerror-converting-circular-structure-to-json) might be helpful.
- If it is not, it is still something specific to vue, because the issue occurs only for Vue2 & Vue3. Angular and React are not affected.

**Comment by AMBudnik** [comment]
Thank you for the research, @sequba 
I'm changing the labels to VUE.

**Comment by maomaobaibai-cloud** [comment]
I'v met the same problem, and i just rewrote the function simpleEqual in node_modules\@handsontable\vue\es\vue-handsontable.mjs line 317 with the code below, and the warning has gone, hope that helpful.
`function simpleEqual(x, y) {
  if (x === y) {
    return true;
  } else if ((typeof x == "object" && x != null) && (typeof y == "object" && y != null)) {
    if (Object.keys(x).length !== Object.keys(y).length) {
      return false;
    }
    for (var prop in x) {
      if (y.hasOwnProperty(prop)) {  
        if (!simpleEqual(x[prop], y[prop])) return false;
      } else {
        return false;
      }
    }
    return true;
  } else {
    return false;
  }
}`

**Comment by 171h** [comment]
+1

**Comment by AMBudnik** [comment]
Hi @GlennMatthys @171h @maomaobaibai-cloud @fsnlarson 

We just merged the pull request that fixes this issue. We'll do our best to push it in the next release.

**Comment by magierg** [comment]
fix verified with 14.1: https://jsfiddle.net/maciej_handsontable/pjdynz90/1/
